### DISCUSSION TITLE
I’d like to confirm a question about PR

### DISCUSSION THREAD

**Discussion by liby** [peaceiris/actions-gh-pages#640] [Q&A] [answered]
I saw  #339 and #497, and noticed this [comment-639546151](https://github.com/peaceiris/actions-gh-pages/issues/339#issuecomment-639546151)

> OK. I will enhance our warning message when this action runs on a forked repo. All users do not know the spec of GitHub Actions' secrets.
> I'm very sorry for taking a long time for this. As a long-service user, I should have noticed your event triggered by a forked repo at first...

My current approach is to skip jobs that require `GITHUB_TOKEN` via `if: ${{ github.ref == 'refs/heads/main' }}`, I see the same code in the repository example.

Although I haven't found the exact documentation, I'm basically sure now that the forked repositories don't read `GITHUB_TOKEN` when submitting a PR. But I see that you quote this sentence:
> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

If I understand correctly: the forked repository is able to read `GITHUB_TOKEN`. I searched the official documentation for this, but I couldn't find any instructions on setting up the forked repository to be allowed to read `GITHUB_TOKEN`.

The only thing relevant is the [`permissions` configuration](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token), but after I set 
```
permissions:
  pull-requests: read
```
I still get [error](https://github.com/KaiserY/trpl-zh-cn/runs/3274100283?check_suite_focus=true). I also found the original text of the above quote:
> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository. The permissions for the GITHUB_TOKEN in forked repositories is read-only. For more information, see "Authenticating with the GITHUB_TOKEN."

If `The permissions for the GITHUB_TOKEN in forked repositories is read-only.`, then why does this [error](https://github.com/KaiserY/trpl-zh-cn/actions/runs/1109977140) still occur?

So I'd like to get a more detailed answer from you, or maybe an official statement that the forked repository doesn't allow reading`GITHUB_TOKEN` related documents, I know this question is a bit off-topic, but I really don't have a choice because I don't know where to get the answer at the moment.

I look forward to hearing from you, have a great day.

**Comment by peaceiris** [comment, marked as answer]
You just forgot the write permission for contents.

```diff
permissions:
+ contents: write
  pull-requests: read
```

**Reply by liby** [reply]
Uh, there doesn't seem to be much explanation of `contents` in the documentation, so can I ask what it actually stands for? I don't even understand the difference between setting `contents: write` and `contents: read`. I really don't know anything about this.

**Reply by liby** [reply]
It doesn't seem to solve the problem. 
https://github.com/KaiserY/trpl-zh-cn/actions/runs/1110393006/workflow

**Reply by peaceiris** [reply]
Ah, I got your situation.

- The run you mentioned has been triggered with the `pull_request` event from your **FORK** `liby:main`
- The run started at the upstream repository, not the fork.
- The generated `GITHUB_TOKEN` with the `pull_request` event from a **FORK** repository does not have any write permission.

That is the reason the `pull_request` event from a **FORK** repository needs the `if: ${{ github.ref == 'refs/heads/main' }}` filter.

If you want to deploy from only a default branch and use the `pull_request` event, please add the filter.

**Reply by liby** [reply]
> The generated GITHUB_TOKEN with the pull_request event from a FORK repository does not have any write permission.
> That is the reason the pull_request event from a FORK repository needs the if: ${{ github.ref == 'refs/heads/main' }} filter.

So in my case, the `GITHUB_TOKEN` generated by the forked repository will only be rejected by the main repository. Therefore, this job can only be skipped, right?

> If you want to deploy from only a default branch and use the pull_request event, please add the filter.

Not fully understood. Are there examples or simple code?

**Reply by peaceiris** [reply]
Yes.

**Reply by liby** [reply]
Something like this?
```yaml
if: ${{github.event_name == 'pull_request' && github.ref == 'refs/heads/master'}}
```

**Reply by peaceiris** [reply]
I think the `if: ${{ github.ref == 'refs/heads/main' }}` filter which is mentioned on the README will satisfy demand of the project.

**Reply by liby** [reply]
Oh, it's really enough to meet my needs.

I just took the following passage as a test question:
> If you want to deploy from only a default branch and use the pull_request event, please add the filter.

and tried to solve it.
